pyroscope.scrape: do not drop __meta labels (#2710)
grafana/alloy: docs/sources/reference/components/pyroscope/pyroscope.write.md

# `pyroscope.write`

> `pyroscope.write` receives performance profiles from other components and forwards them to a series of user-supplied endpoints using [Pyroscope' Push API](/oss/pyroscope/). When forwarding profiles, most labels starting with double underscore (`__`) are dropped before sending the data, with the following exceptions:
> - `__name__` - preserved as it identifies the profile type
> - `__delta__` - preserved as it's required for delta profiles

Reviewer: ```suggestion
`pyroscope.write` receives performance profiles from other components and forwards them to a series of user-supplied endpoints using [Pyroscope' Push API](/oss/pyroscope/).
When `pyroscope.write` forwards profiles, most labels starting with double underscore (`__`) are dropped before the data is sent, with the following exceptions:

* `__name__` is preserved because it identifies the profile type.
* `__delta__`is preserved because it's required for delta profiles.
```

The link to `/oss/pyroscope` lands on https://grafana.com/oss/pyroscope/  Is this really where we want to direct people? Or shoudl we maybe point people at the API docs instead? Maybe here https://grafana.com/docs/pyroscope/latest/reference-server-api/ instead?

Author: I agree we should put a better link to push API, however I'm afraid we don't have the doc for the push API or at least I'm not aware of such docs.

The link you shared is not `Push API`, but `Ingest API` which is doing the same thing, just with different APIS.

The API itself is here https://github.com/grafana/pyroscope/tree/main/api maybe we can link to the source code at least. 

Or remove "Push API" mentioning and the link alltogether

Reviewer: Updated the suggestion

Reviewer 2: I would also say `all labels [...]` instead of `most labels [...]`.

Reviewer: Updated the suggestion

> - `__delta__` - preserved as it's required for delta profiles

Reviewer 3: If we end up specifying what labels aren't dropped, should we mention `__meta` labels as well?
